### DISCUSSION TITLE
column auto completion

### DISCUSSION THREAD

**Discussion by minyoung90** [sqlalchemy/sqlalchemy#7529] [Usage Questions] [answered]
Hi, I am using sqlalchemy==1.4 and sqlalchmy2-stubs==0.0.2-alpha.19

I have this code
```
stmt = (
                sa.select(models.IComEmailAuthTicket)
                .where(models.IComEmailAuthTicket.email == confirm_info.email)
                .order_by(sa.desc(models.IComEmailAuthTicket.send_time))
                .limit(1)
            )
obj_ticket = orm_session.execute(stmt).scalar_one_or_none()
```

mypy doesn't know about what obj_ticket's type is.
I want to use auto-completion for obj_ticket's column

I tried like this
```
obj_ticket: models.IComEmailAuthTicket  = orm_session.execute(stmt).scalar_one_or_none()
```
but, scalar_one_or_none() returns Optional[Any], so mypy doesn't allow it.

is there any way to solve it?

**Comment by CaselIT** [comment, marked as answer]
Hi,

Ince the return value can be None, wouldn't the correct type be `Optional[models.IComEmailAuthTicket]`? I think mypy would be fine with that.

At the moment select does not type the return value. Work in ongoing to try to support that feature for v2

**Reply by minyoung90** [reply]
Thanks!

Optional is fine for me, but '# type: ignore' is more good to me for waiting v2
